drizzlepac/haputils/pipeline_graphics.py

```diff
@@ -0,0 +1,411 @@
+be to define the arrow heads as 2 additional segments starting at the endpoint
+of the vector. The new segments would then be added to the data points to be
+plotted by Segment to appear as an arrow head on each segment.
+
+
+
+# Auto-scaling computation for Quiver plot from matplotlib.
+# https://github.com/matplotlib/matplotlib/blob/master/lib/matplotlib/quiver.py#L666-673
+ 
+# self.N : number of points to be plotted
+# amean : mean of vector lengths
+# self.span : width of plot
+
+            sn = max(10, math.sqrt(self.N))
+            if self.Umask is not ma.nomask:
+                amean = a[~self.Umask].mean()
+            else:
+                amean = a.mean()
+            # crude auto-scaling
+            # scale is typical arrow length as a multiple of the arrow width
+            scale = 1.8 * amean * sn / self.span
+
+Example from bokeh.org on how to create a tabbed set of plots:
+
+.. code:: python
+
+    from bokeh.io import output_file, show
+    from bokeh.models import Panel, Tabs
+    from bokeh.plotting import figure
+
+    output_file("slider.html")
+
+    p1 = figure(plot_width=300, plot_height=300)
+    p1.circle([1, 2, 3, 4, 5], [6, 7, 2, 4, 5], size=20, color="navy", alpha=0.5)
+    tab1 = Panel(child=p1, title="circle")
+
+    p2 = figure(plot_width=300, plot_height=300)
+    p2.line([1, 2, 3, 4, 5], [6, 7, 2, 4, 5], line_width=3, color="navy", alpha=0.5)
+    tab2 = Panel(child=p2, title="line")
+
+    tabs = Tabs(tabs=[ tab1, tab2 ])
+
+    show(tabs)
+
+
+"""
+
+# -------------------------------------------------------------------------------
+# Generate the Bokeh plot for the pipeline astrometric data.
+#
+HOVER_COLUMNS = ['gen_info.imgname',
+                 'gen_info.instrument',
+                 'gen_info.detector',
+                 'gen_info.filter',                
+                 'header.DATE-OBS',
+                 'header.RA_TARG',
+                 'header.DEC_TARG',
+                 'header.GYROMODE',
+                 'header.EXPTIME',
+                 'fit_results.aligned_to']
+                   
+TOOLTIPS_LIST = ['EXPNAME', 'INSTRUMENT', 'DET', 'FILTER', 
+                  'DATE', 'RA', 'DEC', 'GYRO', 'EXPTIME',
+                  'ALIGNED_TO']
+INSTRUMENT_COLUMN = 'full_instrument'
+
+RESULTS_COLUMNS = ['fit_results.rms_x',
+                   'fit_results.rms_y',
+                   'fit_results.xsh',
+                   'fit_results.ysh',
+                   'fit_results.rot',
+                   'fit_results.scale',
+                   'fit_results.rot_fit',
+                   'fit_results.scale_fit',
+                   'fit_results.nmatches',
+                   'fit_results.skew']
+
+RESIDS_COLUMNS = ['residuals.x',
+                  'residuals.y',
+                  'residuals.ref_x',
+                  'residuals.ref_y']
+TOOLSEP_START = '{'
+TOOLSEP_END = '}'
+    
+FIGURE_TOOLS = graph_utils.FIGURE_TOOLS_BASE
```

Comment:
For the new interface, this should be removed from this file.  We decided to have a consistent set of tools and not allow the user to add more tools on-the-fly.  This entry should be ignored by the constructor, and I see I added a comment.  However, I did not eliminate the code!  I will fix.  In any case, this line of code should be eliminated.

Reply:
Removed in 057ba8a.